File: frontend/src/model/Book.ts
class Book {
	public id: number;
	public title: string;
	public author: string;
	public description: string;
	public price: number;
	public quantity: number;
	public categoryId: number;
	/**
	 * image use base64
	 */
	public image: string;

	public constructor(
		id: number,
		title: string,
		author: string,
		description: string,
		quantity: number,
		price: number,
		image: string,
		categoryId: number,
	) {
		this.id = id;
		this.title = title;
		this.author = author;
		this.description = description;
		this.quantity = quantity;
		this.price = price;
		this.image = image;
		this.categoryId = categoryId;
	}
}

export default Book;

File: backend/src/main/java/com/haruma/library/entity/Book.java
package com.haruma.library.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name="book")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@Builder
public class Book {

    @Id
    @Column(name="id", nullable = false, unique = true)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name="title", nullable = false)
    private String title;

    @Column(name="author", nullable = false)
    private String author;

    @Column(name="description", nullable = false)
    private String description;

    @Column(name="quantity", nullable = false)
    private Integer quantity;

    @Column(name="price", nullable = false)
    private BigDecimal price;

    @ManyToOne(cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
    @JoinColumn(name = "category_id")
    @JsonIgnore
    private Category category;

    @Column(name="image", nullable = false)
    private String image;

    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Review> reviewList = new ArrayList<>();

    @OneToMany(mappedBy = "book", cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
    @JsonIgnore
    private List<CartItem> cartItems = new ArrayList<>();

    @OneToMany(mappedBy = "book")
    @JsonIgnore
    private List<Order> orders = new ArrayList<>();

}
